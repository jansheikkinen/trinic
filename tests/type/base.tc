// Base

function main(void) -> void do
  let a: int8,   b: int16,   c: int32,   d: int64,   e: isize = undefined;
  let f: uint8,  g: uint16,  h: uint32,  i: uint64,  j: usize = undefined;
  let k: float8, l: float16, m: float32, n: float64, o: fsize = undefined;
  let p: byte,   q: char,    r: void,    s: bool,    t: any   = undefined;
  let u: identifier = undefined;
end
