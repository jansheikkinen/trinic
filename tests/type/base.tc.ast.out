## PARSER ##
Parsing tests/type/base.tc...

### LEXER ###
Generating tokens from source...
Generated 108 tokens
000:     FUNCTION,     (null), (003, 008)
001:   IDENTIFIER,       main, (003, 013)
002:            (,     (null), (003, 014)
003:         VOID,     (null), (003, 018)
004:            ),     (null), (003, 019)
005:           ->,     (null), (003, 021)
006:         VOID,     (null), (003, 026)
007:           DO,     (null), (003, 029)
008:          LET,     (null), (004, 005)
009:   IDENTIFIER,          a, (004, 007)
010:            :,     (null), (004, 008)
011:         INT8,     (null), (004, 013)
012:            ,,     (null), (004, 014)
013:   IDENTIFIER,          b, (004, 018)
014:            :,     (null), (004, 019)
015:        INT16,     (null), (004, 025)
016:            ,,     (null), (004, 026)
017:   IDENTIFIER,          c, (004, 030)
018:            :,     (null), (004, 031)
019:        INT32,     (null), (004, 037)
020:            ,,     (null), (004, 038)
021:   IDENTIFIER,          d, (004, 042)
022:            :,     (null), (004, 043)
023:        INT64,     (null), (004, 049)
024:            ,,     (null), (004, 050)
025:   IDENTIFIER,          e, (004, 054)
026:            :,     (null), (004, 055)
027:        ISIZE,     (null), (004, 061)
028:            =,     (null), (004, 063)
029:    UNDEFINED,     (null), (004, 073)
030:            ;,     (null), (004, 074)
031:          LET,     (null), (005, 005)
032:   IDENTIFIER,          f, (005, 007)
033:            :,     (null), (005, 008)
034:        UINT8,     (null), (005, 014)
035:            ,,     (null), (005, 015)
036:   IDENTIFIER,          g, (005, 018)
037:            :,     (null), (005, 019)
038:       UINT16,     (null), (005, 026)
039:            ,,     (null), (005, 027)
040:   IDENTIFIER,          h, (005, 030)
041:            :,     (null), (005, 031)
042:       UINT32,     (null), (005, 038)
043:            ,,     (null), (005, 039)
044:   IDENTIFIER,          i, (005, 042)
045:            :,     (null), (005, 043)
046:       UINT64,     (null), (005, 050)
047:            ,,     (null), (005, 051)
048:   IDENTIFIER,          j, (005, 054)
049:            :,     (null), (005, 055)
050:        USIZE,     (null), (005, 061)
051:            =,     (null), (005, 063)
052:    UNDEFINED,     (null), (005, 073)
053:            ;,     (null), (005, 074)
054:          LET,     (null), (006, 005)
055:   IDENTIFIER,          k, (006, 007)
056:            :,     (null), (006, 008)
057:       FLOAT8,     (null), (006, 015)
058:            ,,     (null), (006, 016)
059:   IDENTIFIER,          l, (006, 018)
060:            :,     (null), (006, 019)
061:      FLOAT16,     (null), (006, 027)
062:            ,,     (null), (006, 028)
063:   IDENTIFIER,          m, (006, 030)
064:            :,     (null), (006, 031)
065:      FLOAT32,     (null), (006, 039)
066:            ,,     (null), (006, 040)
067:   IDENTIFIER,          n, (006, 042)
068:            :,     (null), (006, 043)
069:      FLOAT64,     (null), (006, 051)
070:            ,,     (null), (006, 052)
071:   IDENTIFIER,          o, (006, 054)
072:            :,     (null), (006, 055)
073:        FSIZE,     (null), (006, 061)
074:            =,     (null), (006, 063)
075:    UNDEFINED,     (null), (006, 073)
076:            ;,     (null), (006, 074)
077:          LET,     (null), (007, 005)
078:   IDENTIFIER,          p, (007, 007)
079:            :,     (null), (007, 008)
080:         BYTE,     (null), (007, 013)
081:            ,,     (null), (007, 014)
082:   IDENTIFIER,          q, (007, 018)
083:            :,     (null), (007, 019)
084:         CHAR,     (null), (007, 024)
085:            ,,     (null), (007, 025)
086:   IDENTIFIER,          r, (007, 030)
087:            :,     (null), (007, 031)
088:         VOID,     (null), (007, 036)
089:            ,,     (null), (007, 037)
090:   IDENTIFIER,          s, (007, 042)
091:            :,     (null), (007, 043)
092:         BOOL,     (null), (007, 048)
093:            ,,     (null), (007, 049)
094:   IDENTIFIER,          t, (007, 054)
095:            :,     (null), (007, 055)
096:          ANY,     (null), (007, 059)
097:            =,     (null), (007, 063)
098:    UNDEFINED,     (null), (007, 073)
099:            ;,     (null), (007, 074)
100:          LET,     (null), (008, 005)
101:   IDENTIFIER,          u, (008, 007)
102:            :,     (null), (008, 008)
103:   IDENTIFIER, identifier, (008, 019)
104:            =,     (null), (008, 021)
105:    UNDEFINED,     (null), (008, 031)
106:            ;,     (null), (008, 032)
107:          END,     (null), (009, 003)

### AST GENERATION ###
Declaration FUNCTION  main
  Stmt VARDECL
    Arg IDENTIFIER
      Identifier_Arg a
        Type TYPE INT8
      Identifier_Arg b
        Type TYPE INT16
      Identifier_Arg c
        Type TYPE INT32
      Identifier_Arg d
        Type TYPE INT64
      Identifier_Arg e
        Type TYPE ISIZE
    Arg EXPR
      Expr UNDEFINED
  Stmt VARDECL
    Arg IDENTIFIER
      Identifier_Arg f
        Type TYPE UINT8
      Identifier_Arg g
        Type TYPE UINT16
      Identifier_Arg h
        Type TYPE UINT32
      Identifier_Arg i
        Type TYPE UINT64
      Identifier_Arg j
        Type TYPE USIZE
    Arg EXPR
      Expr UNDEFINED
  Stmt VARDECL
    Arg IDENTIFIER
      Identifier_Arg k
        Type TYPE FLOAT8
      Identifier_Arg l
        Type TYPE FLOAT16
      Identifier_Arg m
        Type TYPE FLOAT32
      Identifier_Arg n
        Type TYPE FLOAT64
      Identifier_Arg o
        Type TYPE FSIZE
    Arg EXPR
      Expr UNDEFINED
  Stmt VARDECL
    Arg IDENTIFIER
      Identifier_Arg p
        Type TYPE BYTE
      Identifier_Arg q
        Type TYPE CHAR
      Identifier_Arg r
        Type TYPE VOID
      Identifier_Arg s
        Type TYPE BOOL
      Identifier_Arg t
        Type TYPE ANY
    Arg EXPR
      Expr UNDEFINED
  Stmt VARDECL
    Arg IDENTIFIER
      Identifier_Arg u
        Type IDENTIFIER identifier
    Arg EXPR
      Expr UNDEFINED
